stdlib/public/stubs/Stubs.cpp

```diff
@@ -487,6 +487,12 @@ static const char *_swift_stdlib_strtoX_clocale_impl(
     const char * nptr, T* outResult, T huge,
     T (*posixImpl)(const char *, char **, locale_t)
 ) {
+  if (strcasecmp(nptr, "snan") == 0 || strcasecmp(nptr, "+snan") == 0 ||
```

Comment:
Mostly paranoia.  You require the length of `nptr` to complete the return value anyways.  Computing it earlier and verifying it matches the length of the expected static strings is a more sound check.